src/support.js

```diff
@@ -235,25 +237,40 @@ const supportCommands = () => {
     cy.get('#menuList a[href*="menutype"]:first').click()
     cy.clickToolbarButton('new')
 
-    // Select a type for the new menu item
-    cy.get('.controls > .input-group > .btn').click();
-    cy.get('#menuTypeModal').should('be.visible')
+    // Open iFrame to select menu item type
+    cy.get('body').then(($body) => {
+      if ($body.find('.controls > .input-group > .btn.btn-primary').length > 0) {
+        // Do it in Joomla 4
+        cy.get('.controls > .input-group > .btn.btn-primary').eq(0).click()
+      } else {
+        // Do it in Joomla >= 5 with click the 'Menu Item Type'
+        cy.get('button[data-button-action="select"]').each(($btn) => {
+          const dataModalConfig = $btn.attr('data-modal-config')
+          if (dataModalConfig && dataModalConfig.includes('Menu Item Type')) {
+            cy.wrap($btn).click()
+          }
+        })
+      }
+    })
 
     cy.get('iframe').iframe().find('button').contains(menuCategory).click()
     cy.get('iframe').iframe().find('.accordion-body a').contains(menuItem).click()
 
-    cy.get('#jform_title').click();
-    cy.get('#jform_title').type(menuTitle);
+    cy.get('#jform_title').click()
+    cy.get('#jform_title').type(menuTitle)
 
-    // TODO: Language settings
+    // The language is only configured for multilingual websites where the language selection is visible.
+    cy.get('#jform_language').should('exist').then(($select) => {
```

Comment:
Is it possible to use here something different, than should, because this looks more like an assertion and not a condition.

Reply:
Yes 👍  `should('exist')` simply omitted as `cy.get('#jform_language')` already will wait for the default timeout.

Re-Tested ten times with the fortunately available script `test.cy.js` with actual 'joomla-cypress' main branch on:

* Intel macOS 14.5 Sonoma native with Joomla 4.4-dev
* Intel macOS 14.5 Sonoma docker-based with Joomla 4.4-dev, 5.1-dev, 5.2-dev and 6.0-dev
* Intel x64 Windows 11 Pro Laragon with Joomla 5.1-dev
* Intel x64 Windows 11 Pro WSL 2 docker-based with Joomla 4.4-dev, 5.1-dev, 5.2-dev and 6.0-dev

```
  Running:  test.cy.js                                                                      (1 of 1)
  Test createMenuItem
    ✓ have fresh start with installJoomla (50539ms)
    ✓ can create with minimum arguments - 1st (5944ms)
    ✓ can create with explicit set menu and (unused) language - 2nd (5268ms)
    ✓ can create in different menu - 3rd (9666ms)
    ✓ change to multilingual website and install second language ja-JP (10919ms)
    ✓ still can create with minimum arguments - 4th (6717ms)
    ✓ can create with explicit set menu and using language ja-JP - 第5 (6005ms)
    ✓ can create with third language uk-UA - 6-й (13088ms)
```